src/aliceVision/lensCorrectionProfile/lcp.cpp

```diff
@@ -0,0 +1,929 @@
+        }
+        else if (!p1.perspParams.isEmpty && !p2.perspParams.isEmpty)
+        {
+            pOut.perspParams.FocalLengthX = interpolate<float>(weightLow, p1.perspParams.FocalLengthX, p2.perspParams.FocalLengthX);
+            pOut.perspParams.FocalLengthY = interpolate<float>(weightLow, p1.perspParams.FocalLengthY, p2.perspParams.FocalLengthY);
+            pOut.perspParams.ImageXCenter = interpolate<float>(weightLow, p1.perspParams.ImageXCenter, p2.perspParams.ImageXCenter);
+            pOut.perspParams.ImageYCenter = interpolate<float>(weightLow, p1.perspParams.ImageYCenter, p2.perspParams.ImageYCenter);
+            pOut.perspParams.RadialDistortParam1 = interpolate<float>(weightLow, p1.perspParams.RadialDistortParam1, p2.perspParams.RadialDistortParam1);
+            pOut.perspParams.RadialDistortParam2 = interpolate<float>(weightLow, p1.perspParams.RadialDistortParam2, p2.perspParams.RadialDistortParam2);
+            pOut.perspParams.RadialDistortParam3 = interpolate<float>(weightLow, p1.perspParams.RadialDistortParam3, p2.perspParams.RadialDistortParam3);
+            pOut.perspParams.isEmpty = false;
+        }
+        else
+        {
+            pOut.fisheyeParams.isEmpty = true;
+            pOut.perspParams.isEmpty = true;
+        }
+        break;
+    }
+    }
+}
+
+void LCPinfo::getDistortionParams(const float& focalLength, const float& focusDistance, LensParam& lparam)
+{
+    settingsInfo userSettings;
+    userSettings.ApertureValue = 0.f;
+    userSettings.FocalLength = focalLength;
+    userSettings.FocusDistance = focusDistance;
+
+    int iLow, iHigh;
+    float weightLow;
+    if (search(userSettings, LCPCorrectionMode::DISTORTION, iLow, iHigh, weightLow))
+    {
+        combine(iLow, iHigh, weightLow, LCPCorrectionMode::DISTORTION, lparam);
+    }
+}
+
+void LCPinfo::getVignettingParams(const float& focalLength, const float& aperture, LensParam& lparam)
+{
+    settingsInfo userSettings;
+    userSettings.ApertureValue = aperture;
+    userSettings.FocalLength = focalLength;
+    userSettings.FocusDistance = 0.f;
+
+    int iLow, iHigh;
+    float weightLow;
+    if (search(userSettings, LCPCorrectionMode::VIGNETTE, iLow, iHigh, weightLow))
+    {
+        combine(iLow, iHigh, weightLow, LCPCorrectionMode::VIGNETTE, lparam);
+    }
+}
+
+// Some useful functions when parsing the LCP database
+
+void parseDirectory(const boost::filesystem::path& p, std::vector<boost::filesystem::path>& v)
+{
+    if (boost::filesystem::is_directory(p))
+    {
+        for (auto&& x : boost::filesystem::directory_iterator(p))
+            parseDirectory(x.path(), v);
+    }
+    else if (boost::filesystem::is_regular_file(p) && (boost::filesystem::extension(p) == ".lcp"))
+    {
+        v.push_back(p);
+    }
+}
+
+std::string reduceString(const std::string& str)
+{
+    std::string localStr = str;
+
+    // remove all space
+    localStr.erase(std::remove(localStr.begin(), localStr.end(), ' '), localStr.end());
```

Comment:
Maybe have a loop over characters here?

Reply:
This one looks ok to me. I agree with all the others. Thanks for the review!!

Reply:
The context is that we repeat `localStr.erase(std::remove(...))` maybe 10 times with different characters below this line. It could appear that I suggested to replace localStr.erase with manual erasure, that is not the case :)

Reply:
you mean
```c++
    for(const auto& c : {' ', '/', '.', '_', '-', '*', ',', ';', ':'})
    {
        localStr.erase(std::remove(localStr.begin(), localStr.end(), c), localStr.end());
    }
   // or declare
   // static constexpr std::array<char, 9> charToRemove{' ', '/', '.', '_', '-', '*', ',', ';', ':'};

Reply:
`boost::algorithm::trim_all_if(localStr, boost::algorithm::is_any_of(" /._-*,;:"));`

Reply:
trim_all_if do not fully erase a given character in a string when this character is in the middle of the string.

```
/*! \file
    Defines trim_all algorithms.
    
    Just like \c trim, \c trim_all removes all trailing and leading spaces from a 
    sequence (string). In addition, spaces in the middle of the sequence are truncated
    to just one character. Space is recognized using given locales.

    \c trim_fill acts as trim_all, but the spaces in the middle are replaces with 
    a user-define sequence of character.

    Parametric (\c _if) variants use a predicate (functor) to select which characters
    are to be trimmed.. 
    Functions take a selection predicate as a parameter, which is used to determine 
    whether a character is a space. Common predicates are provided in classification.hpp header.

*/
```